Add features to sample banner

Abban opened the pull request:
We have external people working on these banners so
this adds samples of common features we need in them.

- Add conditional loading example to sample banner.
- Add impression counting to sample banner.
- Add tracking to sample banner.
- Add document on how to create a new banner.
- Add js and css linting.
- Fix linter errors.
- Fix typo in test banner div element id.

timEulitz commented:
@tzhelyazkova 

https://de.wikipedia.org/?banner=WMDE_techwish_voting_2019

I think it might still look a bit bad on some of the MediaWiki skins, I have not tested any of them as of now. Maybe double-check that.

Abban committed:
Add features to sample banner

We have external people working on these banners so
this adds samples of common features we need in them.

- Add conditional loading example to sample banner.
- Add impression counting to sample banner.
- Add tracking to sample banner.
- Add document on how to create a new banner.
- Add js and css linting.
- Fix linter errors.
- Fix typo in test banner div element id.

Abban committed:
Add lint workflow

moiikana committed:
Update package-lock.json

moiikana commented:
rebased because of conflicting package-lock
